test/production/app-dir/ssg-single-pass/ssg-single-pass.test.ts

```diff
@@ -0,0 +1,50 @@
+import { nextTestSetup } from 'e2e-utils'
+import { retry } from 'next-test-utils'
+
+describe('ssg-single-pass', () => {
+  const { next, skipped } = nextTestSetup({
+    skipDeployment: true,
+    files: __dirname,
+  })
+
+  if (skipped) {
+    return
+  }
+
+  it('should only render the page once during build', async () => {
+    const logOccurrences = next.cliOutput.split('home page rendered').length - 1
+    expect(logOccurrences).toBe(1)
```

Comment:
for better readability in case of failures:

```suggestion
    expect(next.cliOutput).toIncludeRepeated('home page rendered', 1)
```

Reply:
TIL that utility 😆